### PR TITLE
remove various typos

### PR DESCRIPTION
**Pull request by scalp42**

### CODE REVIEW

File: consul/structs/structs.go [hunk 1]
```diff
@@ -380,7 +380,7 @@ func (r *KeyRequest) RequestDatacenter() string {
 type KeyListRequest struct {
 	Datacenter string
 	Prefix     string
-	Seperator  string
+	Separator  string
```
Comment (ryanbreen) [inline consul/structs/structs.go]: Question for @armon or @ryanuber or @slackpad: will this rename break any third-parties reliant upon `Seperator`?  Is our contract with those third-parties such that names can break from release to release?

**Comment by ryanbreen** [comment]
These are great fixes, thanks!  I'm happy to merge this.  I just have the one question about how safe it is to rename a field in a struct.

**Comment by scalp42** [comment]
`Seperator` was sed to `Separator` globally, I would think it doesn't break anything but looks like I did. Feel free to close in case, it's a Friday PR :innocent:

**Comment by ryanbreen** [comment]
sed was global to Consul.  I'm thinking about external projects that might rely on the currently defined field name in struct `KeyListRequest`.

**Comment by slackpad** [comment]
Yeah this would break the API. If we wanted to clean it up in a safe way I think we'd need to support both forms and change the docs to be the correct, new way. May not be worth the code complexity, though - the v2 API can fix all these in a legit way.

**Comment by ryanbreen** [comment]
Cool, makes sense.  @scalp42, if you want to update the PR to restrict the changes to errors in comments and docs, it'll be safe to merge.

File: consul/tombstone_gc_test.go [hunk 2]
```diff
@@ -98,7 +98,7 @@ func TestTombstoneGC_Expire(t *testing.T) {
 
 	select {
 	case <-gc.ExpireCh():
-		t.Fatalf("shoudl be reset")
+		t.Fatalf("should be reset")
```
Comment (scalp42) [inline consul/tombstone_gc_test.go]: should I leave this one?

Comment (ryanbreen) [inline consul/tombstone_gc_test.go]: Sure.  I considered saying "comments, docs, and strings," but some of the strings are parsed params that correlate to struct field names, I think.  Stuff like this is fine.

Comment (slackpad) [inline consul/tombstone_gc_test.go]: Yeah anything inside `*_test.go` should be ok.

**Comment by scalp42** [comment]
Should be good now!

File: command/agent/kvs_endpoint.go [hunk 3]
```diff
@@ -91,15 +91,15 @@ func (s *HTTPServer) KVSGet(resp http.ResponseWriter, req *http.Request, args *s
...
 	// we now are forced to check for both spellings
 	var sep string
 	params := req.URL.Query()
 	if _, ok := params["seperator"]; ok {
 		sep = params.Get("seperator")
 	}
-	if _, ok := params["separator"]; ok {
-		sep = params.Get("separator")
+	if _, ok := params["seperator"]; ok {
+		sep = params.Get("seperator")
```
Comment (scalp42) [inline command/agent/kvs_endpoint.go]: this one is a little bit confusing, looks like it's the actual correct spelling.

Comment (ryanbreen) [inline command/agent/kvs_endpoint.go]: That's the outdated diff (that is, it's your first diff).  The spelling in master is `seperator`: https://github.com/hashicorp/consul/blob/e347f20790b81a7a64bff05756900e9294f46892/command/agent/kvs_endpoint.go#L98

**Comment by ryanbreen** [comment]
Thanks!

**Pull request merged**